### PR TITLE
Add improved user profile export functionality

### PR DESCRIPTION
**Pull request by madurangasiriwardena**
### Proposed changes in this pull request

Resolves https://github.com/wso2/product-is/issues/16602
Add new user information providers for 
- basic profile details (deprecating the existing provider)
- linked account details

### When should this PR be merged

Immediately

### Follow up actions
Merge https://github.com/wso2/carbon-identity-framework/pull/4902

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/org.wso2.carbon.identity.user.export.core/src/main/java/org/wso2/carbon/identity/user/export/core/internal/UserProfileExportDataHolder.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        return userInformationProviders;
+    }
+
+    public static void setUserInformationProviders(
+            List<UserInformationProvider> userInformationProviders) {
+
+        UserProfileExportDataHolder.userInformationProviders = userInformationProviders;
+    }
+
+    private static List<UserInformationProvider> userInformationProviders = new ArrayList<>();
```
Comment (Thisara-Welmilla) [inline components/org.wso2.carbon.identity.user.export.core/src/main/java/org/wso2/carbon/identity/user/export/core/internal/UserProfileExportDataHolder.java:49]: Shall we add method comment for these also. There are some other method missing the method comments

File: components/org.wso2.carbon.identity.user.export.core/src/main/java/org/wso2/carbon/identity/user/export/core/internal/UserProfileExportServiceComponent.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+            ConsentInformationProvider consentInformationProvider = new ConsentInformationProvider();
+            ctxt.getBundleContext().registerService(UserInformationProvider.class.getName(), consentInformationProvider,
+                    null);
+
+            SecurityInformationProvider securityInformationProvider = new SecurityInformationProvider();
+            ctxt.getBundleContext().registerService(UserInformationProvider.class.getName(), securityInformationProvider,
+                    null);
+        } catch (Exception e) {
+            LOG.error(e.getMessage(), e);
```
Comment (Thisara-Welmilla) [inline components/org.wso2.carbon.identity.user.export.core/src/main/java/org/wso2/carbon/identity/user/export/core/internal/UserProfileExportServiceComponent.java:81]: What will be the e.getMessage(). In other places, we have error msg as `LOG.error("Error while activating UserProfileExportServiceComponent", e);

File: components/org.wso2.carbon.identity.user.export.core/src/main/java/org/wso2/carbon/identity/user/export/core/internal/service/impl/AbstractUserInformationProvider.java [hunk 3]
```diff
@@ -19,9 +19,38 @@
...
+    protected static final String WSO2_CLAIM_URI = "http://wso2.org/claims/";
+    protected static final String WSO2_IDENTITY_CLAIM_URI = "http://wso2.org/claims/identity/";
+    protected static final String WSO2_RUN_TIME_CLAIM_URI = "http://wso2.org/claims/runtime/";
+
+    protected UserStoreManager getUserStoreManager(int tenantId, String userStoreDomain) throws UserExportException {
+
+        AbstractUserStoreManager userStoreManager;
+        try {
+            RealmService realmService = UserProfileExportDataHolder.getRealmService();
+            UserRealm userRealm = realmService.getTenantUserRealm(tenantId);
```
Comment (Thisara-Welmilla) [inline components/org.wso2.carbon.identity.user.export.core/src/main/java/org/wso2/carbon/identity/user/export/core/internal/service/impl/AbstractUserInformationProvider.java:44]: Is it possible to userRealm to be null?

Comment (madurangasiriwardena) [reply components/org.wso2.carbon.identity.user.export.core/src/main/java/org/wso2/carbon/identity/user/export/core/internal/service/impl/AbstractUserInformationProvider.java:44]: According to the implementation of these methods, there seems to be no possibility. And this seems to be the common practice in handling these 2 objects in the code base.

**Review by Thisara-Welmilla — APPROVED** [review]

**Pull request merged**